react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -85,19 +86,24 @@ const NumberInput = (props) => (
           return(
             <div className="ma__input-number">
               <input {...inputAttr} />
+              {
+                props.unit && <span className="ma__input-number-unit">{props.unit}</span>
```

Comment:
i would add a condition to not render this if the value is null, i.e. when the placeholder is being rendered